### PR TITLE
Blog app - controllers

### PR DESCRIPTION
**Pull request by rahimialijan**
# Blog app - controllers

In this task, I set up UsersController and PostControllers as below:
-  For each URL I created:
    - A route.
    - An action in the correct controller.
    - A view file.

- Fixed all linters errors

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+  get '/users', to: 'users#index', as: 'users'
+
+  # Single user page
+  get '/users/:id', to: 'users#show', as: 'user'
+
+  # All posts by a given user page
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+
+  # Single post page
+  get '/posts/:id', to: 'posts#show', as: 'post'
```
Comment (gemmen29) [inline config/routes.rb]: - [ ] Kindly try to match the given URL in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png) of the project requirement for the `posts#show` action.

Comment (gemmen29) [inline config/routes.rb]: - [ ] [OPTIONAL] Kindly write the routes in this way:
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

as it's cleaner and easier to read and maintain.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @rahimialijan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @rahimialijan 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR 👍 
* Controllers written properly 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**